testing/bin/run_tests.py

```diff
@@ -20,27 +20,50 @@
 )
 sys.path.insert(0, os.environ["_PEX_TEST_PROJECT_DIR"])
 
-from pex.typing import TYPE_CHECKING
+from pex.compatibility import urlparse
+from pex.typing import TYPE_CHECKING, cast
 from testing import devpi, pex_project_dir
 
 if TYPE_CHECKING:
-    from typing import Iterator, Tuple
+    from typing import Iterator, Mapping, Optional, Tuple
 
 
 def iter_test_control_env_vars():
     # type: () -> Iterator[Tuple[str, str]]
     for var, value in sorted(os.environ.items()):
-        if re.search(r"(PEX|PYTHON)", var) and var != "PYTHONHASHSEED":
+        if re.search(r"(PEX|PIP|PYTHON)", var) and var != "PYTHONHASHSEED":
             yield var, value
 
 
 def main():
     # type: () -> int
     parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        "-l",
+        "--log-level",
+        type=lambda arg: arg.upper(),
+        default="INFO",
+        choices=["DEBUG", "INFO", "WARN", "WARNING", "ERROR", "CRITICAL"],
+        help="Set the logging level (case insensitive).",
+    )
     parser.add_argument("--color", default=None, action="store_true", help="Force colored logging.")
     parser.add_argument(
         "--devpi", action="store_true", help="Proxy PyPI through a local devpi server."
     )
+    parser.add_argument(
+        "--require-devpi",
```

Comment:
This is used in a subsequent commit in this PR to ensure the job that builds up the cache is actually populating a devpi-server data dir - the ~whole point.